subworkflows/local/build_references.nf

```diff
@@ -142,7 +143,15 @@ workflow BUILD_REFERENCES {
             !file(params.starfusion_ref_stub_check).exists() || file(params.starfusion_ref_stub_check).isEmpty() )) {
             STARFUSION_BUILD(ch_fasta, ch_gtf, params.fusion_annot_lib, params.species)
             ch_versions = ch_versions.mix(STARFUSION_BUILD.out.versions)
-            ch_starfusion_ref = STARFUSION_BUILD.out.reference
+            if (params.ctatsplicing || params.all) {
```

Comment:
```suggestion
            if (params.ctatsplicing) {
```

Reply:
The params.all is in the outer if clause:

```
 if ((params.starfusion || params.all) &&
            (!file(params.starfusion_ref).exists() || file(params.starfusion_ref).isEmpty() ||
            !file(params.starfusion_ref_stub_check).exists() || file(params.starfusion_ref_stub_check).isEmpty() )) {
            STARFUSION_BUILD(ch_fasta, ch_gtf, params.fusion_annot_lib, params.species)
```

I think that is why the `test_build` is failing. It is not populating the `ch_starfusion_ref` with  `CTATSPLICING_PREPGENOMELIB.out.reference` or `STARFUSION_BUILD.out.reference`

Reply:
Hmm but the ctatsplicing reference creation will not run if `--all` is supplied this way?

Reply:
I think it is difficult to differentiate which is the default here. If `--all` is supplied, should `ch_starfusion_ref` produce `STARFUSION_BUILD.out.reference` or `CTATSPLICING_PREPGENOMELIB.out.reference`?

Before adding your new module `CTATSPLICING_PREPGENOMELIB`, the default for `--all` was to use the `STARFUSION_BUILD.out.reference` but now it will always default to `CTATSPLICING_PREPGENOMELIB.out.reference`.

Reply:
Hi yeah sorry for the confusion:
`CTATSPLICING_PREPGENOMELIB.out.reference` basically is the `STARFUSION` reference with some added extra files to the `ctatsplicing` folder inside of it. So this does not emit a different reference but an expanded reference

Reply:
Then it makes sense to be the default then :) we should update the snapshot of the `test_build` so that it emits these new files.

Reply:
@atrigila I'm not really sure what is going wrong with the `test_build` test but something seems broken in `STARFUSION_BUILD`. Would you have an idea on what could be going wrong?

Reply:
I'll check and get back to you later today. If I don't, please ping me again.

Reply:
Okay great, thank you!

Reply:
I am sorry, I am not sure what's going on. It doesn't seem to happen in other PRs, maybe it was something when merging dev into this branch?

Reply:
Okay thank you! The only thing that changes for `STARFUSION_BUILD` is the configuration name, which shouldn't change much (unless the old configuration wasn't used before this change). I'll investigate a bit further

Reply:
Yes this seems to be the issue. I'll remove the args and that should fix it

Reply:
It was a copy paste error on my part :grin: